containers/Home/Contact/RenderTypeContact.tsx

```diff
@@ -0,0 +1,118 @@
+    }
+}
+
+const HandleRenderTypeOnDesktop: React.FC<RenderTypeProps> = ({ type, linkProps, content }) => {
+    const isMobile = useIsMobile()
+
+    const propStyleCircleMobile = useSpring({ from: { scale: 0, opacity: 0.6 }, scale: 1.2, opacity: 1, reset: true })
+    const [{ scale, opacity }, setPropStyleCircle] = useSpring(() => ({
+        from: { scale: 0, opacity: 0.6 },
+        scale: 1.2,
+        opacity: 1,
+    }))
+    const [flip, set] = useState(false)
+    const propStyleIconMobile = useSpring({ from: { rotateZ: 0 }, rotateZ: 1, reverse: flip, onRest: () => set(!flip) })
+    const [{ rotateZ }, setPropStyleIcon] = useSpring(() => ({
+        from: { rotateZ: 0 },
+        rotateZ: 1,
+        reverse: flip,
+        onRest: () => set(!flip),
+    }))
+    const propStyleCircle = isMobile ? propStyleCircleMobile : { scale, opacity }
+    const propStyleIcon = isMobile ? propStyleIconMobile : { rotateZ }
+
+    const stylesIcon = {
+        transform: propStyleIcon.rotateZ
+            .to({
+                range: [0, 0.25, 0.5, 0.75, 1],
+                output: [0, -20, 20, -10, 0],
+            })
+            .to((rotateZ) => `rotateZ(${rotateZ}deg)`),
+    }
+    const stylesCircle = {
+        transform: propStyleCircle.scale
+            .to({
+                range: [0, 0.25, 0.5, 0.75, 1],
+                output: [1.05, 1.1, 1.2, 1.1, 1.05],
+            })
+            .to((scale) => `scale(${scale})`),
+        opacity: propStyleCircle.opacity
+            .to({
+                range: [0, 0.25, 0.5, 0.75, 1],
+                output: [1, 0.6, 0.3, 0.6, 1],
+            })
+            .to((opacity) => opacity),
+    }
+
+    const handleIcon = (type: string) => {
+        switch (type) {
+            case 'skype':
+                return <Image src="/images/skype.svg" alt="gitIcon" width={18} height={18} />
+            case 'mail':
+                return <Image src="/images/gmailIcon.svg" alt="gitIcon" width={18} height={18} />
+            case 'phone':
+                return 'call'
+        }
+    }
+    switch (type) {
+        case 'phone':
+        case 'skype':
+        case 'mail':
+            return (
+                <span style={{ position: 'relative' }}>
+                    <StyledWrapCircle style={{ ...stylesCircle }} />
+                    <StyledIcon
+                        style={{ ...stylesIcon }}
+                        className="material-icons"
+                        {...linkProps}
+                        onMouseEnter={() => {
+                            setPropStyleCircle({ scale: 1.2, opacity: 1, reset: true, loop: true })
+                            setPropStyleIcon({ rotateZ: 1, loop: true })
+                        }}
+                        onMouseLeave={() => {
+                            setPropStyleCircle({ scale: 0, opacity: 0.6, reset: false, loop: false })
+                            setPropStyleIcon({ rotateZ: 0, loop: false })
```

Comment:
We should moveout these two handling functions to a separate functions, it would make clearer render structure's view